xcm/pallet-xcm/src/lib.rs

```diff
@@ -230,16 +272,41 @@ pub mod pallet {
 			maybe_notify: Option<(u8, u8)>,
 			timeout: BlockNumber,
 		},
+		/// The query is for an ongoing version notification subscription.
+		VersionNotifier { origin: VersionedMultiLocation, is_active: bool },
 		/// A response has been received.
 		Ready { response: VersionedResponse, at: BlockNumber },
 	}
 
-	/// Value of a query, must be unique for each query.
-	pub type QueryId = u64;
+	#[derive(Copy, Clone)]
+	pub(crate) struct LatestVersionedMultiLocation<'a>(pub(crate) &'a MultiLocation);
+	impl<'a> EncodeLike<VersionedMultiLocation> for LatestVersionedMultiLocation<'a> {}
+	impl<'a> Encode for LatestVersionedMultiLocation<'a> {
+		fn encode(&self) -> Vec<u8> {
+			let mut r = VersionedMultiLocation::from(MultiLocation::default()).encode();
+			r.truncate(1);
+			self.0.using_encoded(|d| r.extend_from_slice(d));
```

Comment:
Huh, if `LatestVersionedMultiLocation` encodes exactly like `VersionedMultiLocation`, what's the reasoning behind this code, i.e. extracting the first byte of a `VersionedMultiLocation` and appending the encoded `MultiLocation` to it?

Reply:
Oh wait nevermind, `LatestVersionedMultiLocation` is a struct, not an enum like `VersionedMultiLocation`.